### PR TITLE
Feature/2806 assets manifest

### PR DESCRIPTION
**Pull request by ivankruchkoff**
## Summary

Addresses issue #2806 

## Relevant technical choices

A few differences from IB:
1. `seed` property added to manifestArgs. It's further in the comments on the ticket but not captured in IB.
2. In `webpack.config.js` changed all entry `keys` for `css` to match the handles in PHP since we match in the manifest based on these handles.
## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: includes/Core/Assets/Stylesheet.php [hunk 1]
```diff
@@ -61,12 +61,21 @@ public function register( Context $context ) {
 		if ( $this->args['fallback'] && wp_style_is( $this->handle, 'registered' ) ) {
 			return;
 		}
+		$src     = $this->args['src'];
+		$version = $this->args['version'];
+
+		$filename = Manifest::get_filename( $this->handle );
+
+		if ( ! is_null( $filename ) ) {
```
Comment (aaemnnosttv) [inline includes/Core/Assets/Stylesheet.php:69]: We can relax this condition a bit. If an empty string were returned for example, that shouldn't trigger the same behavior as if there was a filename.

```suggestion
		if ( $filename ) {
```

File: includes/Modules/Analytics/Advanced_Tracking/Script_Injector.php [hunk 2]
```diff
@@ -54,10 +54,11 @@ public function inject_event_script( $events ) {
 			return;
 		}
 
-		// Get file contents of script and add it to the page, injecting event configurations into it.
-		$filename = 'analytics-advanced-tracking.js';
-		if ( class_exists( '\Google\Site_Kit\Core\Assets\Manifest' ) && isset( Manifest::$assets['analytics-advanced-tracking'] ) ) {
-			$filename = Manifest::$assets['analytics-advanced-tracking'];
+		$filename = Manifest::get_filename( 'analytics-advanced-tracking' );
+		if ( is_null( $filename ) ) {
```
Comment (aaemnnosttv) [inline includes/Modules/Analytics/Advanced_Tracking/Script_Injector.php:58]: Similarly here as well 🙂 
```suggestion
		if ( ! $filename ) {
```

File: webpack.config.js [hunk 3]
```diff
@@ -40,32 +40,45 @@ const projectPath = ( relativePath ) => {
...
+				.map(
+					( key ) =>
+						`"${ key
+							.replace( '.js', '' )
+							.replace( '.css', '' ) }"${ ''.padEnd(
+							maxLen - key.length,
+							' '
+						) } => "${ manifest[ key ] }",`
+				)
+				.join( '\n\t\t' )
```
Comment (aaemnnosttv) [inline webpack.config.js:63]: Each line should only have a single indentation.
```suggestion
				.join( '\n\t' )
```

File: includes/Core/Assets/Script.php [hunk 4]
```diff
@@ -68,19 +68,9 @@ public function register( Context $context ) {
...
-				$handle = str_replace( 'googlesitekit-', '', $this->handle );
-				if ( isset( Manifest::$assets[ $handle ] ) ) {
-					$filename = Manifest::$assets[ $handle ];
-				}
-			}
-		}
+		$filename = Manifest::get_filename( $this->handle );
 
-		if ( ! empty( $filename ) ) {
+		if ( ! is_null( $filename ) ) {
```
Comment (aaemnnosttv) [inline includes/Core/Assets/Script.php:73]:
```suggestion
		if ( $filename ) {
```

File: includes/Core/Assets/Manifest.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * Assets manifest.
+ *
+ * @since 1.15.0
+ * @access private
+ * @ignore
+ */
+class Manifest {
+
+	private static $assets;
```
Comment (aaemnnosttv) [inline includes/Core/Assets/Manifest.php:24]: Please add a doc block.

File: includes/Core/Assets/Manifest.php [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+	public static function get_filename( $handler ) {
```
Comment (aaemnnosttv) [inline includes/Core/Assets/Manifest.php:26]: Please add a doc block.

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @ivankruchkoff – this is really close, just a few details to touch up.

File: includes/Core/Assets/Manifest.php (hunk 5 above)
Comment (ivankruchkoff) [reply includes/Core/Assets/Manifest.php:30]: https://github.com/google/site-kit-wp/pull/4095/commits/325d420950e98519cbe6e1d31604f66373814d82

File: includes/Core/Assets/Script.php (hunk 4 above)
Comment (ivankruchkoff) [reply includes/Core/Assets/Script.php]: https://github.com/google/site-kit-wp/pull/4095/commits/01d9abe1d5ed557ff063b7758cb9cd8ac1bc1fd1

File: includes/Core/Assets/Stylesheet.php (hunk 1 above)
Comment (ivankruchkoff) [reply includes/Core/Assets/Stylesheet.php]: https://github.com/google/site-kit-wp/pull/4095/commits/01d9abe1d5ed557ff063b7758cb9cd8ac1bc1fd1

File: includes/Modules/Analytics/Advanced_Tracking/Script_Injector.php (hunk 2 above)
Comment (ivankruchkoff) [reply includes/Modules/Analytics/Advanced_Tracking/Script_Injector.php]: https://github.com/google/site-kit-wp/pull/4095/commits/c8d815593f99020e7988a6cd053bc87c43ee08ac

File: webpack.config.js (hunk 3 above)
Comment (ivankruchkoff) [reply webpack.config.js]: https://github.com/google/site-kit-wp/pull/4095/commits/dc71e03de34fb6360f90c28611d8c4f0bd5ddcf2

File: webpack.config.js [hunk 7]
```diff
@@ -40,32 +40,45 @@ const projectPath = ( relativePath ) => {
...
+		);
+
+		return content;
+	},
+};
+
 const manifestTemplate = `<?php
 /**
- * Class Google\\Site_Kit\\Core\\Assets\\Manifest
  *
```
[the remark this answers is not in the document]
Comment (ivankruchkoff) [reply webpack.config.js]: https://github.com/google/site-kit-wp/pull/4095/commits/dc71e03de34fb6360f90c28611d8c4f0bd5ddcf2

File: webpack.config.js [hunk 8]
```diff
@@ -40,32 +40,45 @@ const projectPath = ( relativePath ) => {
...
  * @package   Google\Site_Kit
```
[the remark this answers is not in the document]
Comment (ivankruchkoff) [reply webpack.config.js]: https://github.com/google/site-kit-wp/pull/4095/commits/dc71e03de34fb6360f90c28611d8c4f0bd5ddcf2

File: webpack.config.js [hunk 9]
```diff
@@ -40,32 +40,45 @@ const projectPath = ( relativePath ) => {
...
- *
- * @since 1.15.0
- * @access private
- * @ignore
- */
-class Manifest {
-
-	public static $assets = array(
+return array(
 		{{assets}}
```
[the remark this answers is not in the document]
Comment (ivankruchkoff) [reply webpack.config.js]: https://github.com/google/site-kit-wp/pull/4095/commits/dc71e03de34fb6360f90c28611d8c4f0bd5ddcf2

File: includes/Core/Assets/Manifest.php [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+	private static $assets;
+
+	/**
+	 * Gets the filename for a given handle.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $handle  Script or stylesheet handle.
+	 */
+	public static function get_filename($handle ) {
```
Comment (aaemnnosttv) [inline includes/Core/Assets/Manifest.php:39]: Odd that PHPCS didn't flag this 🤷‍♂️ 

```suggestion
	public static function get_filename( $handle ) {
```

**Review by aaemnnosttv — APPROVED** [review]
:shipit:

**Pull request merged**